akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -49,6 +47,11 @@ sealed trait ModeledHeader extends HttpHeader with Serializable {
   protected def companion: ModeledCompanion[_]
 }
 
+private[headers] sealed trait RequestHeader extends ModeledHeader { override def renderInRequests = true }
+private[headers] sealed trait ResponseHeader extends ModeledHeader { override def renderInResponses = true }
+private[headers] sealed trait RequestResponseHeader extends RequestHeader with ResponseHeader
+private[headers] sealed trait SyntheticHeader extends ModeledHeader
```

Comment:
Yeah, I thought about that as well but didn't want to introduce even more disruption with this PR.
And, from the user perspective it's nice to have to remember only a single "rule" on how to find a header:
just look in the `...headers` package.